prow/cmd/sinker/main.go

```diff
@@ -124,15 +135,44 @@ func main() {
 		config:        cfg,
 	}
 
-	// Clean now and regularly from now on.
-	for {
-		start := time.Now()
-		c.clean()
-		logrus.Infof("Sync time: %v", time.Since(start))
-		if o.runOnce {
-			break
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+		// Clean now and regularly from now on.
+		for {
+			start := time.Now()
+			c.clean()
+			logrus.Infof("Sync time: %v", time.Since(start))
+			if o.runOnce {
+				break
+			}
+			time.Sleep(cfg().Sinker.ResyncPeriod)
 		}
-		time.Sleep(cfg().Sinker.ResyncPeriod)
+	}()
+
+	if !o.runOnce {
```

Comment:
Serving the prometheus data when we run once is actually not the end of the world, so in the name of clarity for the code here let's just set up the server at the beginning of `main` and not make this complicted

Reply:
Done.